### ISSUE TITLE
[Build] onnxruntime infer dynamic scale imgs in windows

### ISSUE THREAD

**Issue by TwinkleStarst** [microsoft/onnxruntime#19075]
### Describe the issue

I encountered an error when calling the onnx model using onnxruntime on Windows, but it can run on Linux.  The programming language I am using is Python. The onnx model is converted on Linux. The input size of the model is dynamic(1*3*h*w). 

### Target platform

windows

### Build script

```python
import os
import cv2
import onnxruntime
import torch

from AutoVia.util.file_util import load_file
from AutoVia.util.json_util import save_json_fun


#加载ONNX模型
def load_model():
    model_path = r"E:\VIA\VIA-model-0109-dd.onnx"
    so = onnxruntime.SessionOptions()
    so.add_session_config_entry('session.load_model_format', 'ONNX')
    # session = onnxruntime.InferenceSession(model_path, sess_options=so, providers=["CPUExecutionProvider"])
    session = onnxruntime.InferenceSession(model_path, sess_options=so)

    return session


#推理
def onnx_predict(img_path, img_format, session):
    images = load_file(img_path)
    new_json = []

    for img in images:
        fn = os.path.basename(img)
        im = cv2.imread(img, -1)

        if img_format == "L":
            if len(im.shape) != 2:
                im = cv2.cvtColor(im, cv2.COLOR_BGR2GRAY)
            inf_im = torch.as_tensor(im.astype("float32"))
        else:
            if len(im.shape) == 2:
                im = cv2.cvtColor(im, cv2.COLOR_GRAY2BGR)
            inf_im = torch.as_tensor(im.astype("float32").transpose(2, 0, 1)[None])
        input_name = session.get_inputs()[0].name
        output_names = [node.name for node in session.get_outputs()]
        input = {input_name: inf_im.cpu().numpy()}

        output = session.run(output_names, input)
        new_json.append(output)

    return new_json


if __name__ == '__main__':

    session = load_model()
    img_path = r"E:\VIA\QtDesigner\AutoVia\inf_img-1"
    img_format = "RGB"
    results = onnx_predict(img_path, img_format, session)
    save_json_fun(results, img_path, "inf_json")
```

### Error / output

onnxruntime.capi.onnxruntime_pybind11_state.RuntimeException: [ONNXRuntimeError] : 6 : RUNTIME_EXCEPTION : Non-zero status code returned while running Add node. Name:'/backbone/Add_1' Status Message: D:\a\_work\1\s\onnxruntime\core/providers/cpu/math/element_wise_ops.h:560 onnxruntime::BroadcastIterator::Append axis == 1 || axis == largest was false. Attempting to broadcast an axis by a dimension other than 1. 15 by 16
![image](https://github.com/microsoft/onnxruntime/assets/63721796/fc88f17e-263e-45de-8db5-611b4654e0d1)
![错误提示](https://github.com/microsoft/onnxruntime/assets/63721796/e505e2b5-e08d-443c-86fd-94b5295171f0)

**Comment by xadupre** [comment]
It seems the onnx model is incorrect. How did you build it? Did you try to run check_model or shape_inference (from onnx package) on it?
